manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -176,22 +178,24 @@ public Long getProcessorsCount(String bridgeId, String customerId) {
 
     @Override
     public void deleteProcessor(String bridgeId, String processorId, String customerId) {
-        List<ConnectorEntity> connectorEntitiesToBeDeleted = updateProcessorConnectorForDeletionOnDB(bridgeId, processorId, customerId);
-        connectorEntitiesToBeDeleted.forEach(c -> {
-            LOGGER.info("Firing deletion event for entity: " + c);
-            eventBus.requestAndForget(Events.CONNECTOR_DELETED_EVENT, c);
-        });
+        Processor processor = doDeleteProcessor(bridgeId, processorId, customerId);
+
+        workManager.schedule(processor);
     }
 
     @Transactional
-    public List<ConnectorEntity> updateProcessorConnectorForDeletionOnDB(String bridgeId, String processorId, String customerId) {
+    protected Processor doDeleteProcessor(String bridgeId, String processorId, String customerId) {
         Processor processor = processorDAO.findByIdBridgeIdAndCustomerId(processorId, bridgeId, customerId);
-        processor.setStatus(BridgeStatus.DEPROVISION);
+        processor.setStatus(ManagedResourceStatus.DEPROVISION);
 
-        LOGGER.info("Processor with id '{}' for customer '{}' on bridge '{}' has been marked for deletion", processor.getId(), processor.getBridge().getCustomerId(),
+        LOGGER.info("Processor with id '{}' for customer '{}' on bridge '{}' has been marked for deletion",
+                processor.getId(),
+                processor.getBridge().getCustomerId(),
                 processor.getBridge().getId());
 
-        return connectorService.deleteConnectorIfNeeded(processor);
+        connectorService.deleteConnectorEntity(processor);
```

Comment:
This now flags connectors as `DEPROVISION` and the applicable Worker removes them.